src/web/src/main/java/tupperdate/web/facade/chats/ChatFacadeImpl.kt

```diff
@@ -1,38 +1,146 @@
 package tupperdate.web.facade.chats
 
+import io.ktor.http.*
+import tupperdate.web.facade.accounts.AccountId
+import tupperdate.web.facade.recipes.toRecipe
 import tupperdate.web.model.Result
+import tupperdate.web.model.accounts.Notification
+import tupperdate.web.model.accounts.NotificationRepository
 import tupperdate.web.model.chats.ChatRepository
+import tupperdate.web.model.chats.Conversation
+import tupperdate.web.model.chats.toModelNewMessage
+import tupperdate.web.model.map
 import tupperdate.web.model.profiles.User
+import tupperdate.web.model.profiles.UserRepository
+import tupperdate.web.model.recipes.RecipeRepository
 
 class ChatFacadeImpl(
-    private val chats: ChatRepository,
+    private val chatRepository: ChatRepository,
+    private val userRepository: UserRepository,
+    private val recipeRepository: RecipeRepository,
+    private val notificationRepository: NotificationRepository,
 ) : ChatFacade {
 
     override suspend fun readAll(
         user: User,
     ): Result<List<Chat>> {
-        TODO("Not yet implemented")
+        val convs: List<Conversation> = when (val res = chatRepository.readAll(user)) {
+            is Result.Ok -> res.result
+            else -> return res.map { listOf() }
+        }
+
+        val chats = convs.map { conv ->
+            val chatUser =
+                when (val res = userRepository.read(user = User(AccountId(conv.theirId)))) {
+                    is Result.Ok -> res.result
+                    else -> return res.map { listOf() }
+                }
+            val lastMessage =
+                when (val res = chatRepository.readMessages(user = user, userId = conv.theirId)) {
+                    is Result.Ok -> res.result
+                    else -> return res.map { listOf() }
+                }.getOrNull(0)
```

Comment:
We probably don't want to scan through the whole message set of a user each time, but rather only take the latest directly.